scripts/simple_frap.py

```diff
@@ -87,46 +115,50 @@ def save_results(conn, image, values):
     image.linkAnnotation(map_ann)
 
 
-# Delete-old-annotations
-def delete_old_annotations(conn, image, namespace):
-    to_delete = []
-    for ann in image.listAnnotations(ns=namespace):
-        to_delete.append(ann.id)
-    if len(to_delete) > 0:
-        print("Deleting old annotations", to_delete)
-        conn.deleteObjects('Annotation', to_delete, wait=True)
-
+# Step 8 - Save the plot
+def save_plot(conn, image, plot_filename):
+    """
+    Save the plot to OMERO
+    :param conn: The BlitzGateway
+    :param image: The image
+    :param plot_filename: The path to the plot image
+    :return: Nothing
+    """
+    pil_img = Image.open(plot_filename)
+    np_array = np.asarray(pil_img)
+    red = np_array[::, ::, 0]
+    green = np_array[::, ::, 1]
+    blue = np_array[::, ::, 2]
+    plane_gen = iter([red, green, blue])
+    conn.createImageFromNumpySeq(plane_gen, plot_filename, sizeC=3,
+                                 dataset=image.getParent())
+
+
+# Step 9 - Wrap it all up
+def analyse(conn, image_id, channel_name):
+    # Step 2 - Load iamge
+    img = conn.getObject("Image", image_id)
+    # -
+    ci = get_channel_index(img, channel_name)
+    shape_id = get_ellipse_roi(conn, img)
+    values = get_mean_intensities(conn, img, ci, shape_id)
+    plot(values, 'plot.png')
+    save_values(conn, img, values)
+    save_plot(conn, img, 'plot.png')
 
-# Disconnect
-def disconnect(conn):
-    conn.close()
 
-
-# main
 def main():
-    # Collect user credentials
     try:
-        host = input("Host [wss://workshop.openmicroscopy.org/omero-ws]: ") or 'wss://workshop.openmicroscopy.org/omero-ws'  # noqa
-        username = input("Username [trainer-1]: ") or 'trainer-1'
+        hostname = "wss://workshop.openmicroscopy.org/omero-ws"
+        username = input("Username: ")
         password = getpass("Password: ")
-        dataset_id = input("Dataset ID [2391]: ") or '2391'
-        channel_label = input("Channel Label [GFP]: ") or 'GFP'
-
-        # Connect to the server
-        conn = connect(host, username, password)
-
-        # Load the images container in the specified dataset
-        images = load_images(conn, dataset_id)
-
-        if len(images) == 0:
-            print("No images in dataset")
-            return
-
-        images = images[:1]
-        analyze(conn, images, channel_label)
+        image_id = int(input("Image ID: "))
```

Comment:
having a default is helpful when running session.
This makes things easier without affecting the generality of the script